src/campaigns/positions/campaignPositions.controller.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Controller,
+  UsePipes,
+  Logger,
+  Get,
+  Post,
+  Put,
+  Param,
+  ParseIntPipe,
+  Delete,
+  Body,
+  UseGuards,
+  HttpCode,
+  HttpStatus,
+} from '@nestjs/common'
+import { ZodValidationPipe } from 'nestjs-zod'
+import { CampaignPositionsService } from './campaignPositions.service'
+import { CreateCampaignPositionSchema } from './schemas/CreateCampaignPosition.schema'
+import { CampaignOwnerOrAdminGuard } from '../guards/CampaignOwnerOrAdmin.guard'
+import { UpdateCampaignPositionSchema } from './schemas/UpdateCampaignPosition.schema'
+
+@Controller('campaigns/:id/positions')
+@UseGuards(CampaignOwnerOrAdminGuard)
+@UsePipes(ZodValidationPipe)
+export class CampaignPositionsController {
+  private readonly logger = new Logger(CampaignPositionsController.name)
+
+  constructor(
+    private readonly campaignPositionsService: CampaignPositionsService,
+  ) {}
+
+  @Get()
+  findByCampaign(@Param('id', ParseIntPipe) id: number) {
+    return this.campaignPositionsService.findByCampaign(id)
```

Comment:
Maybe this should be named `findByCampaignId` to make it more explicit that the `id` param is indeed a campaign id? 🤔